### PR TITLE
Index warm up optimization

### PR DESCRIPTION
**Pull request by alexkorotkikh**
- [x] Remove condition to not ingest future key together with current key
- [x] Add a metric to monitor number of ingestions of future keys
- [x] Adjust a future time bucket calculation to be exactly the same as for the current time bucket
- [x] Ignore Cassandra Datastore unhealthy check
- [x] Log in case of JVM deadlock threads unhealthy check

Tested on staging: https://grafana-staging.zalando.net/d/0Tw3ROxWz/kdb-staging-dashboard?orgId=1&from=now-1h&to=now&refresh=1m

### CODE REVIEW

**Comment by alex-on-java** [comment]
👍

**Comment by alexkorotkikh** [comment]
👍

File: src/main/java/org/kairosdb/core/health/HealthCheckResource.java [hunk 1]
```diff
@@ -18,98 +21,93 @@
...
-		responseBuilder.header("Access-Control-Allow-Origin", "*");
-		responseBuilder.header("Pragma", "no-cache");
-		responseBuilder.header("Cache-Control", "no-cache");
-		responseBuilder.header("Expires", 0);
-
-		return (responseBuilder);
-	}
+public class HealthCheckResource {
+    private final HealthCheckService m_healthCheckService;
+    public static final Logger logger = LoggerFactory.getLogger(CassandraDatastore.class);
```
Comment (pitr) [inline src/main/java/org/kairosdb/core/health/HealthCheckResource.java]: should be `HealthCheckResource.class`

File: src/main/java/org/kairosdb/core/health/HealthCheckResource.java [hunk 2]
```diff
@@ -18,98 +21,93 @@
...
+     * Health check
+     *
+     * @return 204 if healthy otherwise 500
+     */
+    @GET
+    @Produces(MediaType.APPLICATION_JSON + "; charset=UTF-8")
+    @Path("check")
+    public Response check() {
+        for (HealthStatus healthCheck : m_healthCheckService.getChecks()) {
+            if ("Datastore-Query".equals(healthCheck.getName())) {
```
Comment (pitr) [inline src/main/java/org/kairosdb/core/health/HealthCheckResource.java]: use `DatastoreQueryHealthCheck.NAME` instead

**Comment by alexkorotkikh** [comment]
👍

**Comment by tfrauenstein** [comment]
👍

**Comment by rajatparida86** [comment]
👍

**Pull request merged**